### PR TITLE
[gatsby-source-mongodb] added support for extra connection string params

### PR DESCRIPTION
**Pull request by lcostea**
added support for MongoDB Atlas with the extra parameters: replicaSet, ssl & authSource
it will not affect existing plugin connections to mongodb, when the extra params are missing
if accepted I will add the needed documentation

### CODE REVIEW

File: packages/gatsby-source-mongodb/src/gatsby-node.js [hunk 1]
```diff
@@ -119,3 +120,23 @@ function createNodes(
...
+    connectionSuffix =  (connectionSuffix ? connectionSuffix + "&authSource=" : "authSource=") + pluginOptions.authSource;
+  }
+
+  if(connectionSuffix) {
+    return "?" + connectionSuffix;
+  }
+  
+  return "";
+
+}
```
[the remark this answers is not in the document]
Comment (lcostea) [reply packages/gatsby-source-mongodb/src/gatsby-node.js]: I was thinking about this before, maybe I can separate these query string params in a separate object inside options, something like:
```
        extraParams : {
          replicaSet: "replica-shard-01",
          ssl: "true",
          authSource: "admin"
        }
```
This way I can call `stringify` directly on that extraParams object.

[the remark this answers is not in the document]
Comment (pieh) [reply packages/gatsby-source-mongodb/src/gatsby-node.js]: this does look better, yeah! it also will allow users to use any "niche" query fields that are needed for their setup without changes to plugin

[the remark this answers is not in the document]
Comment (lcostea) [reply packages/gatsby-source-mongodb/src/gatsby-node.js]: yeap, great, will do the changes :)

**Comment by m-allanson** [comment]
Thanks @lcostea 👍 It looks like there are some [eslint warnings](https://travis-ci.org/gatsbyjs/gatsby/jobs/394267529) that are causing the tests to fail at the moment. Could you fix those up? Run `npm run lint` to see the warnings locally.

If you're also able to add docs for the new params then I think this will be ready to merge :)

**Comment by jlengstorf** [comment]
Hiya @lcostea! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**